### PR TITLE
Fixes clock app and its settings

### PR DESCRIPTION
**Pull request by nicomiguelino**
### Checklist

- [x] Remove unnecessary fields from `screenly.yml`
- [x] Rename `disable_analytics` to `enable_analytics` to avoid double negatives (which hinders code readability).
- [x] Displays the time in AM/PM or 24-hour format depending on the current location specified in the metadata.
- [x] Make sure that uploading the Edge App is successful.
- [x] Update `README.md`.
- [x] Add a setting for overriding the locale (`override_locale`).
- [x] Add validation for the `override_timezone` setting.
- [x] Add validation for the `override_locale` setting.

### Screenshots

Using coordinates for Paris, FR (as of 10:08 PM, PST):

image

### CODE REVIEW

**Comment by nicomiguelino** [comment]
@vpetersson, @sergey-borovkov, @korvyashka
It seems that old settings are not wiped out after an `upload`.

**Comment by vpetersson** [comment]
> @vpetersson, @sergey-borovkov, @korvyashka It seems that old settings are not wiped out after an `upload`.

Yes, this is correct. I think this is should probably be a manual operation.

**Comment by nicomiguelino** [comment]
@vpetersson I've pushed a new commit, but I still need to sort out a few things:
- GIven the current implementation, will override_timezone still be relevant? If yes, how will I make it play well with the current logic?
- Now I'm added a code that does API calls to an endpoint that does reverse geocoding, just to the the country. I saw a JS library alternative, but it's bulky when built (~38MB, see [geojson-places](https://www.npmjs.com/package/geojson-places) for details).
- Initially, when loading the page, the page shows the time in 12-hour format. It will take a while (depending on how long it will take the API and function calls to finish) for the localized time to be displayed.

File: edge-apps/clock/screenly.yml [hunk 1]
```diff
@@ -1,15 +1,15 @@
+---
 app_id: ''
```
Comment (sergey-borovkov) [inline edge-apps/clock/screenly.yml:2]: Please remove this line as well everywhere

Comment (sergey-borovkov) [reply edge-apps/clock/screenly.yml:2]: Just a clarification - app_id can't be empty in the manifest. It should be not present at all or have a proper non-empty id.

**Comment by vpetersson** [comment]
> GIven the current implementation, will override_timezone still be relevant? If yes, how will I make it play well with the current logic?

Yes, the override_timezone is still relevant. It will just be used instead of the location.

> Now I'm added a code that does API calls to an endpoint that does reverse geocoding, just to the the country. I saw a JS library alternative, but it's bulky when built (~38MB, see [geojson-places](https://www.npmjs.com/package/geojson-places) for details).

Let's use the library.  I know it sounds like a lot, but it will become more reliable.

>Initially, when loading the page, the page shows the time in 12-hour format. It will take a while (depending on how long it will take the API and function calls to finish) for the localized time to be displayed.

Right, so this is solved by using the library too.

**Comment by nicomiguelino** [comment]
@vpetersson, @korvyashka
I've pushed a new commit and tested it on my browser locally.
However, after adding new files, I was not able to upload my Edge App due to the following error:

```
Failed to upload edge app: Asset processing error: Asset "country-locale-map.1.18.5.min.js". Error: "File type not supported.".
```

I tried to run `file --mime-type country-locale-map.1.18.5.min.js`, and I got:

```
country-locale-map.1.18.5.min.js: application/octet-stream
```

@korvyashka, would it be possible to have this kind of `.js` file supported? For context, I built [`country-locale-map`](https://www.npmjs.com/package/country-locale-map) using Webpack.

**Comment by nicomiguelino** [comment]
@vpetersson, @MuratUrsavas, @sergey-borovkov
I already finalized the changes for this app. I've pushed newer commits. Requesting another code review...

- Updated `README.md`.
- Didn't add some complex error-handling for the `override_timezone`. The library responsible for getting the current time (based on the given TZ) does the job of displaying the default time if the value provided is invalid.

**Comment by nicomiguelino** [comment]
@MuratUrsavas, would like to follow up on this one... Thanks.

**Review by iamplugged — APPROVED** [review]
LGTM

**Comment by nicomiguelino** [comment]
@vpetersson, would like to request for this PR to be merged so that @salmanfarisvp can start fresh.
(The commits can be squashed.)

**Pull request merged**